### PR TITLE
Add missing features

### PR DESCRIPTION
**Pull request by 0nkery**

### CODE REVIEW

File: src/mqtt.rs [hunk 1]
```diff
@@ -310,12 +316,30 @@ impl OutgoingEventProperties {
 
 #[derive(Debug, Serialize)]
 pub struct OutgoingRequestProperties {
-    method: &'static str,
+    method: String,
+    correlation_data: String,
+    response_topic: String,
+    #[serde(flatten)]
+    authn: Option<AuthnProperties>,
```
Comment (manifest) [inline src/mqtt.rs]: We also need a `#[serde(skip_serializing_if = "Option::is_none")]` here

File: src/mqtt.rs [hunk 2]
```diff
@@ -310,12 +316,30 @@ impl OutgoingEventProperties {
...
 }
 
 impl OutgoingRequestProperties {
-    pub fn new(method: &'static str) -> Self {
-        Self { method }
+    pub fn new(
+        method: String,
+        response_topic: String,
+        authn: Option<AuthnProperties>,
```
Comment (manifest) [inline src/mqtt.rs]: It better to put optional arguments last.

File: src/mqtt.rs [hunk 3]
```diff
@@ -360,7 +384,7 @@ impl<T, P> OutgoingMessage<T, P>
 where
     T: serde::Serialize,
 {
-    fn new(payload: T, properties: P, destination: Destination) -> Self {
+    pub fn new(payload: T, properties: P, destination: Destination) -> Self {
```
Comment (manifest) [inline src/mqtt.rs]: The `new` constructor should be private. `multicast` and `unicast` should be used in apps instead.

File: src/mqtt.rs (hunk 1 above)
Comment (0nkery) [reply src/mqtt.rs]: `flatten` is not working with `skip_serializing_if` but `flatten` as-is works as intended.

**Pull request merged**